modules/statistics/php/statistics_dd_site.class.inc

```diff
@@ -25,9 +25,47 @@ namespace LORIS\statistics;
  */
 class Statistics_DD_Site extends statistics_site
 {
-
     var $query_criteria = '';
     var $query_vars     = array();
+    /**
+     * Checking user's permission
+     *
+     * @param \User $user The user whose access is being checked
+     *
+     * @return bool
+     */
+    function _hasAccess(\User $user) : bool
+    {
+        // TODO (#6742): Create a permission specific to statistics
+        $hasAccessToAllProfiles = $user->hasPermission('access_all_profiles');
+        if ($hasAccessToAllProfiles) {
+            return true;
+        }
+
+        $hasCenterPermission = false;
+
+        // TODO (#6742): There are no means of setting permissions per site
+        // for a given user right now (see #6742 for more info)
+
+        // If a CenterID is passed in the request, check if the user has the
+        // data_entry permission at the site/center specified by CenterID.
+        if (!empty($_REQUEST['CenterID'])) {
+            $hasCenterPermission = $user->hasCenterPermission(
+                'data_entry',
+                intval($_REQUEST['CenterID'])
+            );
+        } else {
+            // FIXME (#6742): For the short term the user will be granted access
+            // if they have permission for a minimum of one of the centers
+            foreach ($user->getCenterIDs() as $centerID) {
```

Comment:
Comment from 6 days ago.. not sure why GitHub removed it:

I don't understand this check.. under what situations would the user have the centerID returned from getCenterIDs but not have permission to the centerID?

Reply:
@racostas can you comment on @driusan 's question ?

Reply:
Additional information can be found here: https://github.com/aces/Loris/issues/6742.
Part of the comment has been migrated in an issue.

Reply:
Hi, thanks for bring this back. This code is a bit old now I think. I will take a look but sure there are lot of things could be refactored/fixed.